internal/mikrotik/types.go

```diff
@@ -7,6 +7,7 @@ type Config struct {
 	Username      string `env:"MIKROTIK_USERNAME,notEmpty"`
 	Password      string `env:"MIKROTIK_PASSWORD,notEmpty"`
 	SkipTLSVerify bool   `env:"MIKROTIK_SKIP_TLS_VERIFY" envDefault:"false"`
+	Comment       string `env:"MIKROTIK_COMMENT" envDefault:"Managed by ExternalDNS"`
```

Comment:
Honestly, I am not a huge fan of putting the comment configuration here, as this struct is meant to store the authentication details for the RouterOS API, i.e. host, port, user, password etc. and this has nothing to do with authentication.

That being said, I don't necessarily know where it would make more sense to have this.

Can you explain a bit more on the use-case and why this is a needed feature?

Reply:
I just came back from holiday and I'll be able to look into #14 some more.

To be fully honest, I am not too fond of the idea of implementing configurable comments. They're an internal mechanism of the app and changing them can have unintended consequences such as losing track of the records that were managed by external dns prior to changing the comments and then crashing while attempting to delete/recreate them.

I don't foresee a reason to change these to anything else in the future on my end either. They're there for identification purposes only and they're rather suggestive as to who has the ownership/responsibility over the record.

Also in my (fairly limited) research whole working on this, I didn't see other external dns providers implementing something like this either.

Reply:
> I am fine with making the comments configurable, but I don't think providing a default comment is necessary. We can handle that on a per-record basis via ProviderSpecific configuration. Does that work for you?

Are you suggesting the app wouldn't set any comment by default at all? If so, on one hand I think it makes a lot of sense. As you say, they're not necessary functionally and less is more maybe.

However on a selfish note, I personally do need external-dns to add consistent comments and since I have no need to customize it beyond that, I'd end up spamming the same annotation everywhere. I also couldn't do [stuff like this](https://github.com/cosandr/external-dns-provider-mikrotik/commit/11fee78d29def99717957807dfad4b262cee77e9) which surprisingly speeds this up quite noticeably and makes my "lock it down as much as possible" self happy.

Reply:
> Are you suggesting the app wouldn't set any comment by default at all? If so, on one hand I think it makes a lot of sense. As you say, they're not necessary functionally and less is more maybe.

Yes, no comment by default sounds good to me. I added them at the start just to validate everything is good and I forgot them in, I guess :sweat_smile:

> However on a selfish note, I personally do need external-dns to add consistent comments and since I have no need to customize it beyond that, I'd end up spamming the same annotation everywhere. 
Sure, we can add that default config. I would also like to keep the `Managed by External DNS` comment on my end if possible 😆 

> I also couldn't do [stuff like this](https://github.com/cosandr/external-dns-provider-mikrotik/commit/11fee78d29def99717957807dfad4b262cee77e9) which surprisingly speeds this up quite noticeably and makes my "lock it down as much as possible" self happy.
Eh, on this one I don't really agree. Like yeah, sure, I see how it can speed up things, but since the comment is configurable per-endpoint, I don't see how we can implement something like this

Reply:
OK, I'm rebasing this on `feat/support-all-dns-record-types` as well FYI.

Reply:
What do you think of something like this to extract some default values from the environment and then just reference that?

https://github.com/mirceanton/external-dns-provider-mikrotik/commit/82826e8b776b5988779602252ba97db63724ea06

Reply:
That doesn't work, you need to call `env.Parse` for it to actually be populated. You've probably noticed that by now though. I'm thinking of letting you work on this at least until you merge `feat/support-all-dns-record-types`, I doubt it's all that productive for me to do anything at this point while you're restructuring and figuring out what goes where.

If you disagree let me know, but otherwise feel free to pick whatever you want from my PRs and close them if the functionality was added some other way.